### DISCUSSION TITLE
502 server error, after using Image tag with next version ^12.1.5

### DISCUSSION THREAD

**Discussion by Duwini** [vercel/next.js#37869] [Help]
Verify canary release
- [ ] I verified that the issue exists in the Next.js canary release

Operating System: 
Platform: Windows
Arch: x64
Version: 21H2

Binaries: 
Node: 16.14.2
npm: 8.5.0

Relevant packages:
next: ^12.1.5
react: ^17.0.2
react-dom: ^17.0.2

### *What browser are you using? (if relevant)*
Chrome Version 102.0.5005.115 (Official Build) (64-bit)

### *How are you deploying your application? (if relevant)*
next start

### *Describe the Bug*
The img tag is replaced using the Image tag as shown below. However, when I navigate through pages I am getting Error 502. I am assuming the reason behind this is Image tag integration because without the Image tag it worked without any Error.

![2](https://user-images.githubusercontent.com/104137235/174726526-e87ee8b1-f560-4e76-ac43-4da79a3150d6.png)

This is the result obtained after using Image tag in the above specified environment. 

![1](https://user-images.githubusercontent.com/104137235/174727867-ca7d4802-b585-477a-b73f-95e9e9f83639.png)

formats: ["image/webp"]

![3](https://user-images.githubusercontent.com/104137235/174746052-a2bc5bf9-1d57-4b73-9476-5fbd2ef06c9b.png)

### **Expected Behavior**
I expect the images on my deployed site to load properly.

### **To Reproduce**
Use a Next.js Image component with any remote URL src, then deploy the app.

**Comment by icyJoseph** [comment]
Is this not related to Cloudflare being down? For about two hours this morning? Do you see the same issue if you build and run production locally?

**Reply by Duwini** [reply]
Yesterday also I got this issue. Locally it works without an issue, however, only in the server, this issue appears.

**Reply by icyJoseph** [reply]
I remember, but cannot find that there was some issue with caching images, do you have any logs? I think the cache was removing them as they were used, and it caused the server to crash. Not sure in which version this was happening though.

> Btw, `priority="true"`, is not needed, just saying `priority` is fine (shorthand for boolean true).

It is very strange that the entire server would crash down like this, because of an image.

I am not sure what could the issue be, other than perhaps an image with malformed data? Again, do you have any logs from the server?

**Reply by Duwini** [reply]
I have used Kubernetes and have a node js backend. Also, there were no logs. When I remove the Image tags it works on the server well.

**Reply by icyJoseph** [reply]
Sorry, but I can't find anything 12.1.5 related, that could point us in the right direction. Without a reproduction demo/repository, or a look into the shape of `data` looks like, or some logs to look at, we can only guess.

One last thing I'd try is to install `sharp` as a dependency, https://nextjs.org/docs/messages/sharp-missing-in-production, I seem to recall that some other package parsed warnings (which is printed when sharp's not present) in the stdout and halted the process with exit code 0. Again, this is just a guess.

**Reply by Duwini** [reply]
I added the `NEXT_SHARP_PATH`, then I did not get the Error 502. However, the images did not load. Again, the img tags I have used load well.

**Reply by icyJoseph** [reply]
MM that's not really telling us much. A regular image tag just fetches from a network. The next/image tag does some optimization work on the server, and caches images for later usage. We ought to assume the problem is during the optimization phase, but without logs we are poking at the dark here.

**Reply by Duwini** [reply]
So @icyJoseph I removed all the image tags and I went step by step. Like I added the image tag on one page and checked in the dev. It went on without crashing however when I added the image tag to the images coming from strapi it got crashed. This is how I have used strapi images..
![Screenshot (20)](https://user-images.githubusercontent.com/104137235/175299588-5e92a54e-7444-4d48-8368-f877ed91dd02.png)

**Reply by icyJoseph** [reply]
Mm that was clever! I'm out of the office for the week, but I think you probably are up to something here.

I'd guess that one of those short circuit operations is resulting in undefined, or boolean.

**Reply by Duwini** [reply]
Hi @icyJoseph, using CSS styling inside the image tag is not an issue right? Regarding the 502 Error

![Screenshot (22)](https://user-images.githubusercontent.com/104137235/176098003-dda061c1-1c26-478f-9237-4f290df9e28b.png)

**Reply by icyJoseph** [reply]
No, I wouldn't think so...

Does the `STRAPI_BASE_URL` end with `/`, I am thinking in the placeholder case it might resolve to something like, `https://api.domain.com//placeholder.png`, or perhaps it does that on the thumbnail url? Is the thumbnail url, relative?

**Reply by Duwini** [reply]
This how I have integrated it 

![5](https://user-images.githubusercontent.com/104137235/176156295-6dc3515f-40f9-461d-a1ec-86dd36295936.png)

Also, do you think I have to use `getServerSideProps`?

**Reply by icyJoseph** [reply]
No, this should work fine, unless it is failing at build time? 

Everything else looks fine though. Perhaps it does fail to do `next build` when you deploy, and that's why you get a 502? because nothing is there to intake requests. That sound possible.

Is everything available to render these images during build time?

**Reply by Duwini** [reply]
Hi, @icyJoseph I finally got this running without the Error 502 actually. 

- I resized the Images that use the `Image` tag (all below 50 kB)
- Added `NEXT_SHARP_PATH` 
- Downgraded next version - 12.1.4 (I think 12.1.6 works too)
- Strapi images are kept in the `img` tag
- Also local images are coming from the S3 bucket now (to speed up image loading)

**Reply by icyJoseph** [reply]
Aha, but so the Strapi images couldn't be loaded by the `next/image` tag after all? That's very odd, but sure, I think if this works, then it is fine. 

Good job! :tada: :tada:
